pkg/network/node/pod_test.go

```diff
@@ -404,10 +410,14 @@ func TestDirectPodUpdate(t *testing.T) {
 
 	podTester := newPodTester(t, "update", socketPath)
 	podManager := newDefaultPodManager()
+	kubeClient := fake.NewSimpleClientset()
 	podManager.podHandler = podTester
+	podManager.podInformers = kinformers.NewSharedInformerFactory(kubeClient, defaultPodInformerResyncPeriod)
+	podManager.listerSynced = podManager.podInformers.Core().V1().Pods().Informer().HasSynced
+	ch := make(chan struct{})
 	_, clusterCIDR, _ := net.ParseCIDR("1.2.0.0/16")
 	_, serviceCIDR, _ := net.ParseCIDR("172.30.0.0/16")
-	err = podManager.Start(tmpDir, "1.2.3.0/24", []common.ParsedClusterNetworkEntry{{ClusterCIDR: clusterCIDR, HostSubnetLength: 8}}, serviceCIDR, string(configv1.AWSPlatformType))
+	err = podManager.Start(ch, tmpDir, "1.2.3.0/24", []common.ParsedClusterNetworkEntry{{ClusterCIDR: clusterCIDR, HostSubnetLength: 8}}, serviceCIDR, string(configv1.AWSPlatformType))
```

Comment:
Do you need to add the generated Pod object here to the podManager's fake client?

Reply:
@squeed can u elaborate here more I am not following what u want to add ?